pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -249,7 +249,7 @@ func (c *Cluster) makeDeployment(nodeName string, devices []rookalpha.Device, se
 					DNSPolicy:          DNSPolicy,
 					InitContainers: []v1.Container{
 						{
-							Args:            []string{"ceph", "osd", "init"},
+							Args:            []string{"ceph", "--ceph-version-name", c.cephVersion.Name, "osd", "init"},
```

Comment:
Instead of adding this to the args, you could add it to the env vars. The var `ROOK_CEPH_VERSION_NAME` will be translated by cobra to map to the arg `--ceph-version-name`. It has helpful to keep the rook command line options more manageable.